.github/workflows/sonar.yml

```diff
@@ -0,0 +1,47 @@
+name: Sonar Cloud
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  sonar_cloud:
+    name: Sonar Cloud
+    runs-on: [self-hosted, i-dont-know]
```

Comment:
@Felixoid which self-hosted runner should I use here?

Reply:
I picked `builder` 🤷‍♂️

Reply:
the builder looks like a proper choice!